### PR TITLE
Fixed broken testcases.

### PR DESCRIPTION
**Pull request by jeroendk**
I am targeting this branch, because this is a bugfix that is backwards compatible.
## Changelog

```markdown

## Subject
Fixed code style issues for Travis CI.

### CODE REVIEW

**Comment by jeroendk** [comment]
@greg0ire How should I handle all the extra classes that are included in the test classes, for example:
```
namespace Sonata\OrderBundle\Tests\Entity;


class OrderTest extends BaseOrder
{
    protected $id;

    public function getId()
    {
        return $this->id;
    }
}

/**
 * @author Vincent Composieux <vincent.composieux@gmail.com>
 */
class BaseOrderTest extends TestCase
{
```

Should I move OrderTest to a separate file so it will pass the psr4 check?

**Comment by greg0ire** [comment]
Yes, and this one is very recent, here is my attempt for this on the admin bundle: https://github.com/sonata-project/SonataAdminBundle/pull/5397

**Comment by jeroendk** [comment]
Yes, was not done yet :-)

**Comment by greg0ire** [comment]
I think I have fixed most issues in #536 , but I had to disallow Buzz 1.16.1, maybe you can allow it again, but fix this in a better way?

**Comment by jeroendk** [comment]
@greg0ire Nice! 

> but I had to disallow Buzz 1.16.1, maybe you can allow it again, but fix this in a better way?

Like i did in my first commit of this pull request?

Should i close this one and wait for #561 to be merged?

**Comment by greg0ire** [comment]
> Like i did in my first commit of this pull request?

Oh I did not notice ^^ 

> Should i close this one and wait for #561 to be merged?

#561 targets master, so no. Please make FlintCI and Travis pass

**Comment by quisse** [comment]
Hi @jeroendk, are you planning on continuing this pr? Would be great help to mine 😁

File: tests/Component/Customer/User.php [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+
+namespace Sonata\Component\Tests\Customer;
+
+class User
+{
+    public function getId()
+    {
+        return 1;
+    }
+}
```
Comment (OskarStark) [inline tests/Component/Customer/User.php]: Please add newline

**Comment by jeroendk** [comment]
@quisse Was a bit busy but will try to do so soon!

**Comment by quisse** [comment]
@jeroendk great, thanks!

**Comment by jeroendk** [comment]
@greg0ire Any idea why these tests fail? They run fine in my local environment.
```
1) Sonata\Component\Tests\Payment\CheckPaymentTest::testSendbank
Expectation failed for method name is equal to 'sendRequest' when invoked 1 time(s).
Method was expected to be called 1 times, actually called 0 times.

2) Sonata\Component\Tests\Payment\PassPaymentTest::testPassPayment
Expectation failed for method name is equal to 'sendRequest' when invoked 1 time(s).
Method was expected to be called 1 times, actually called 0 times.
```

**Comment by greg0ire** [comment]
What's more concerning to me is the number of commits in your PR and their authors… let me rebase it for you…

**Comment by greg0ire** [comment]
I do reproduce the issue locally BTW (I did not check before the rebase)… you should run `git fetch --all && git reset --hard 03c9fa34 && composer install`

**Comment by greg0ire** [comment]
I just merged another PR, let me rebase again.

**Comment by greg0ire** [comment]
I do reproduce the issue locally BTW (I did not check before the rebase)… you should run git fetch --all && git reset --hard 94d2017b && composer update -v

**Comment by jeroendk** [comment]
@greg0ire Sorry about that, thought i rebased it already. But I can reproduce it now, thanks!

**Comment by jeroendk** [comment]
@greg0ire So it seems that my fix does not work with lower versions of Buzz. Can i disallow version 1.16.1 for the 2.x branch like you did for the master? Or is that not backwards compatible?

**Comment by greg0ire** [comment]
Is it just the tests or is the source code also broken?

**Comment by jeroendk** [comment]
If I check the code in PassPayment + CheckPayment they both rely on `$response->getContent();` The Guzzle Reponse (that is returned in the newer version) object however does not have that function. It uses the `Psr\Http\Message\MessageInterface` that provides `getBody();` instead. 

So i guess the source code is broken as well.

**Comment by greg0ire** [comment]
Then please do disallow the version

**Comment by greg0ire** [comment]
Thanks @jeroendk !

**Comment by jeroendk** [comment]
@greg0ire is it desirable to make a patch for the master branch that works with the new Buzz bundle and drops the support for older version?

**Comment by greg0ire** [comment]
Yes it is! Please do that :)

**Pull request merged**